app/controllers/invitations_controller.rb

```diff
@@ -0,0 +1,24 @@
+class InvitationsController < ApplicationController
+  before_action :require_user
+  
+  def new
+    @invitation = Invitation.new
+  end
+  
+  def create
+    @invitation = Invitation.new(invitation_params.merge!(inviter_id: current_user.id))
+    if @invitation.save
+      AppMailer.send_invitation_email(@invitation).deliver
+      flash['success'] = "An invitation has been sent to #{@invitation.recipient_name}."
+      redirect_to new_invitation_path
+    else
+      flash['danger'] = "Your invitation was not sent."
```

Comment:
Whenever using `render`, you need to use `flash.now`. `flash` will survive the next request as well.